test/e2e/e2e_test.go

```diff
@@ -128,9 +138,21 @@ var _ = Describe("When BYOH joins existing cluster [PR-Blocking]", func() {
 			}
 		}
 
-		os.Remove(agentLogFile1)
-		os.Remove(agentLogFile2)
-		os.Remove(ReadByohControllerManagerLogShellFile)
-		os.Remove(ReadAllPodsShellFile)
+		err := os.Remove(agentLogFile1)
+		if err != nil {
```

Comment:
I don't think we need `if` statements. Directly we can assert `Expect(err).NotTo(HaveOccurred())`

Reply:
Same for all `err != nil` checks in test files

Reply:
i feel these checks should flag the error but not fail the test cases as these are house cleaning activities, but if we feel very strongly about erring as test failure then we will have to revisit these checks
- check if file exists [seen these being already removed before these house cleaning funcs are invoked and this introduces further flakiness if we enforce test failure on these failures
- clean/remove only if the files are not already removed

Reply:
Good point. These don't have to fail the test. I'm +1 on `clean/remove only if the files are not already removed`. I can open a new issue for this. wdyt?